packages/math/src/SphericalHarmonics3.ts

```diff
@@ -0,0 +1,268 @@
+import { IClone } from "@oasis-engine/design";
+import { Color } from "./Color";
+import { Vector3 } from "./Vector3";
+
+/**
+ * Use SH3 to represent irradiance environment maps efficiently, allowing for interactive rendering of diffuse objects under distant illumination.
+ * @remarks
+ * https://graphics.stanford.edu/papers/envmap/envmap.pdf
+ * http://www.ppsloan.org/publications/StupidSH36.pdf
+ * https://google.github.io/filament/Filament.md.html#annex/sphericalharmonics
+ */
+export class SphericalHarmonics3 implements IClone {
+  private static _basisFunction = [
+    0.282095, //  1/2 * Math.sqrt(1 / PI)
+
+    -0.488603, // -1/2 * Math.sqrt(3 / PI)
+    0.488603, //  1/2 * Math.sqrt(3 / PI)
+    -0.488603, // -1/2 * Math.sqrt(3 / PI)
+
+    1.092548, //  1/2 * Math.sqrt(15 / PI)
+    -1.092548, // -1/2 * Math.sqrt(15 / PI)
+    0.315392, //  1/4 * Math.sqrt(5 / PI)
+    -1.092548, // -1/2 * Math.sqrt(15 / PI)
+    0.546274 //  1/4 * Math.sqrt(15 / PI)
+  ];
+
+  private static _convolutionKernel = [
+    3.141593, //  PI
+    2.094395, // (2 * PI) / 3,
+    0.785398 //   PI / 4
+  ];
+
+  private _coefficients: Float32Array = new Float32Array(27);
```

Comment:
coefficients should public and provide for developers like `Vector3`'s `x`,`y`,`z`.

Reply:
+1